extensions/amp-intaker-chat-widget/0.1/widget.js

```diff
@@ -0,0 +1,570 @@
+  let chatUrlHash = ''; //window[INTAKER_CHAT_URL] || window[INTKER_CHAT_URL];
+  let url = ''; //atob(chatUrlHash);
+  let useQA = window['USE_INTAKER_QA'];
+  let frameContainer, closeBtn, restartBtn, lunchBtn, frame;
+  let api = '';
+  const cssUrl =
+    (useQA
+      ? 'https://intakerclientqa.azurewebsites.net'
+      : 'https://intaker.co') + '/dist/chat-widget.min.css';
+  let directLink = '';
+  const isDesktop = !isMobile();
+  let cookieName = 'INTAKER_CHAT_WIDGET_';
+  const cookieValue = {};
+  const DEFAULT_AVATAR =
+    'https://intaker.blob.core.windows.net/dashboard/a6.png';
+  let dingSound = null;
+  let launcherContainer = null;
+  let isUniqueVisit = false;
+  let externalUrl = '';
+  const originalTitle = document.title || '';
+  let isAMP = false;
+  let platform = null;
+  let Templates = window['IntakerWidgetTemplates'];
+  let chatUniqueId = null;
+  let manualOpenedChat = false;
+
+  /**
+   *
+   * @param {Element} el
+   * @param {string} name
+   * @return {Element}
+   */
+  function addClass(el, name) {
+    el.classList.add(name);
+    return el;
+  }
+
+  /**
+   *
+   * @param {Element} el
+   * @param {string} name
+   * @return {Element}
+   */
+  function removeClass(el, name) {
+    el.classList.remove(name);
+    return el;
+  }
+
+  /**
+   *
+   * @param {string} url
+   */
+  function loadCss(url) {
+    const link = document.createElement('link');
+    // link.id    = 'chatter-bot-css';
+    link.rel = 'stylesheet';
+    link.type = 'text/css';
+    link.href = url;
+    link.media = 'all';
+    (document.getElementsByTagName('head')[0] || document.body).appendChild(
+      link
+    );
+  }
+
+  /**
+   *
+   * @param {string} string
+   */
+  function addToBody(string) {
+    let tmp = document.createElement('div');
+    tmp./*OK*/ innerHTML = string;
+    document.body.appendChild(tmp.firstChild);
+    tmp = null;
+  }
+
+  /**
+   *
+   */
+  function closeFrame() {
+    exitPreview();
+    visitor(false);
+    chatIsActive = false;
+    removeClass(frameContainer, 'chatter-bot-frame-container-active');
+    removeClass(document.body, 'chatter-bot-body-noscroll');
+    removeClass(launcherContainer, 'chatter-bot-frame-container-active');
+    hideEl(frameContainer);
+    document.title = originalTitle;
+  }
+
+  /**
+   *
+   */
+  function openFrame() {
+    showEl(frameContainer);
+    addClass(frameContainer, 'chatter-bot-frame-container-active');
+    addClass(launcherContainer, 'chatter-bot-frame-container-active');
+    if (!isDesktop) {
+      if (previewChatIsActive) {
+        addClass(frameContainer, 'preview');
+        addClass(launcherContainer, 'preview');
+      } else {
+        exitPreview();
+      }
+    }
+  }
+
+  /**
+   *
+   * @param {string} url
+   * @param {JsonObject} data
+   * @param {function(?): undefined} success
+   * @return {XMLHttpRequest}
+   */
+  function postAjax(url, data, success) {
+    const xhr = window.XMLHttpRequest
```

Comment:
This should be reimplemented in terms of AMP's XHR service.

Reply:
done